snap/info.go

```diff
@@ -1292,10 +1292,14 @@ var SanitizePlugsSlots = func(snapInfo *Info) {
 	panic("SanitizePlugsSlots function not set")
 }
 
+func ReadInfo(name string, si *SideInfo) (*Info, error) {
+	return ReadInfoFromMountPoint(name, MountDir(name, si.Revision), MountFile(name, si.Revision), HooksDir(name, si.Revision), si)
+}
+
 // ReadInfo reads the snap information for the installed snap with the given
 // name and given side-info.
-func ReadInfo(name string, si *SideInfo) (*Info, error) {
-	snapYamlFn := filepath.Join(MountDir(name, si.Revision), "meta", "snap.yaml")
+func ReadInfoFromMountPoint(name, mountPoint, mountFile, hooksDir string, si *SideInfo) (*Info, error) {
```

Comment:
this also needs its own doc comment then, also afaiu it should compute hooksDir like it computes snapYamlFn instead of being passed in

Reply:
Done